### DISCUSSION TITLE
file_out replacement

### DISCUSSION THREAD

**Discussion by ghost** [ropensci/targets#321] [Help]
Hello!

I remember drake having file_out, and I know targets has a format = "file" to replace the file_in. I was wondering if there was an equivalent in targets or it's become redundant?

Kind Regards,

Marc

**Comment by wlandau** [comment]
All files in `targets` are dynamic files: supply `format = "file"` to `tar_target()` and return a character vector of the file/directory paths you want to track (one hash for all of them). Whether you are describing input files or output files, the interface is the same: 

* Input files: https://books.ropensci.org/targets/files.html#external-input-files
* Output files: https://books.ropensci.org/targets/files.html#external-output-files

`drake` also had dynamic files, but it barely worked when combined with all the other features: https://books.ropensci.org/drake/plans.html#dynamic-files.

There is no `file_out()` in `targets` because new `drake` users always found it confusing because `file_out()` needs to know the literal file path in advance.

**Reply by ghost** [reply]
Thank you,

So if I would like to save a file in my cache, and export it, I should use two separate targets is what I'm understanding.

**Reply by wlandau** [reply]
I usually do. If the duplication takes up too much storage, you could write a single target to run the computation and save the external file all in one go (avoids saving to `_targets/objects/`).
